### ISSUE TITLE
Make it clearer how to set memory limits, make it easier to set them.

### ISSUE THREAD

**Issue by christophfink** [r5py/r5py#35]
> 1. Setting memory
> Setting Java memory `TransportNetwork(java_params=['-Xmx2G'])`.  What's the best approach to allows users allocate emory to Java? In `r5r`, users need to set one time this at the start of the R session, before loading the r5r library. I believe there should be better ways to do this in Python.

This is splitting up issue #25 into smaller pieces (re: https://github.com/r5py/r5py/issues/25#issuecomment-1122901350 )

**Comment by christophfink** [comment]
There are several ways of setting the memory limit:

1. create a config file (see https://r5py.readthedocs.io/en/latest/installation.html#configuration ) and set `max-memory`
2. once we have a CLI executable, set its command line parameter `--max-memory`
3. extend `sys.argv` before importing r5py (see below)

All three methods accept the same range of parameters: 

>         Memory limit for the JVM running R5.
>         Use % as a suffix to specify a share of total RAM;
>         M, G, T to specify MiB, GiB, or TiB, respectively.
>         Values without suffix are interpreted as bytes.
>         Values are rounded to the closest MiB.

(compare https://github.com/r5py/r5py/blob/main/src/r5py/util/memory_footprint.py#L19 )

-----
I agree that this has to go more prominently into the documentation, as an own header + section, most likely. I would maybe suggest to *not* put it into the basic usage/vignette/primer, but rather together with other more advanced topics.
-----

example for (3):
```
import sys
sys.argv.extend(['--max-memory', "5G"])
import r5py
```

**Comment by christophfink** [comment]
What we need to discuss, though, is the default value, and whether it’s sensible on all systems. 

I now defined it to default to 80% of (all) RAM, unless there’s less than 2GiB left after that. Note that this value is the **maximum** the JVM tries to allocate, and I tried to find a middle way between using as much as possible (thinking about virtual machines with 100s of GiB RAM), but leaving enough for the OS and other things that might be running (min. 2GiB). Most Linux, Windows, and MacOS installations have a concept of swap/swap files, so unless the user is running a second memory-heavy application in parallel, we should be fine.

**Comment by christophfink** [comment]
> I agree that this has to go more prominently into the documentation, as an own header + section, most likely. I would maybe suggest to _not_ put it into the basic usage/vignette/primer, but rather together with other more advanced topics.

Still reiterating the importance of documenting this well.

**Comment by wklumpen** [comment]
I do think it should be set up as a warning/exclamation in the documentation for any "Getting Started" documents, since setting the appropriate memory is sort of a big fundamental choice in terms of performance and experience with the system.

To play devil's advocate: One advantage of making the user set an explicit memory (or at least requiring some sort of call/action to set it up) is that they are made aware that we are sitting on top of a fairly beefy java machine running underneath. My worry is that the 80%/2GB rule might create some problems down the road that are more slippery to diagnose.

**Comment by christophfink** [comment]
That’s a very valid point. Let’s add a warning to the `basic-usage.ipynb`, so everybody *should* have read it.

**Comment by christophfink** [comment]
This was maybe addressed sufficiently in issue #90 /PR #96
